src/vstt/stats.py

```diff
@@ -579,7 +705,134 @@ def get_acceleration(
 def _spatial_error(
     mouse_position: np.ndarray, target: np.ndarray, target_radius: float
 ) -> float:
+    """
+    at timeout linear distance from cursor to target - target radius
+
+    :param mouse_position: The array of mouse positions
+    :param target: The position of the target
+    :param target_radius: radius of the target
+    :return: the distance between the end point of the movement to the center of the target - target radius
+    """
     if mouse_position.size < 1:
         return 0
     spatial_error = xydist(mouse_position[-1], target) - target_radius
     return max(spatial_error, 0)
+
+
+def get_first_movement_index(
+    mouse_times: np.ndarray,
+    mouse_positions: np.ndarray,
+    to_target_num_timestamps_before_visible: int,
+) -> int | None:
+    """
+    get index of the first movement in mouse_times
+
+    :param mouse_times: The array of timestamps
+    :param mouse_positions: The array of mouse positions
+    :param to_target_num_timestamps_before_visible: The index of the first timestamp where the target is visible
+    :return: the first movement index in mouse_times
+    """
+    if (
+        mouse_times.shape[0] != mouse_positions.shape[0]
+        or mouse_times.shape[0] == 0
+        or mouse_times.shape[0] < to_target_num_timestamps_before_visible
+    ):
+        return None
+    i = 0
+    while xydist(
+        mouse_positions[0], mouse_positions[i]
+    ) < docs.conf.epsilon and i + 1 < len(mouse_times):
+        i += 1
+    return i
+
+
+def _movement_time_at_peak_velocity(
+    mouse_times: np.ndarray,
+    mouse_positions: np.ndarray,
+    to_target_num_timestamps_before_visible: int,
+) -> float | None:
+    """
+    get the time from first movement to the peak velocity
+
+    :param mouse_times: The array of timestamps
+    :param mouse_positions: The array of mouse positions
+    :param to_target_num_timestamps_before_visible: The index of the first timestamp where the target is visible
+    :return: the time from first movement to the peak velocity
+    """
+    i = get_first_movement_index(
+        mouse_times, mouse_positions, to_target_num_timestamps_before_visible
+    )
+    _, peak_index = _peak_velocity(mouse_times, mouse_positions)
+    return mouse_times[peak_index] - mouse_times[i] if i is not None else None
```

Comment:
Returning `float | None` is good if we are going to do something with the result depending on if it is `None` or not.

But if we are putting the result into our stats table (which I think is the case here?), then it should always be a float, and `np.nan` is what is already used elsewhere for this case, e.g. in `_rmse()`